### DISCUSSION TITLE
How to make database extension inside project with sequel?

### DISCUSSION THREAD

**Discussion by fbayaffo2097** [jeremyevans/sequel#2246] [General]
@jeremyevans Sorry to bother you again today. I've been trying to make a custom extension and I keep getting errors

I have Roda + sequel + Zeitwerk app. 

Inside my sequel.rb I add:

```
require_relative '../../lib/sequel/extensions/my_extension'
```

lib/sequel/extensions/my_extensions.rb:

```
module Sequel
  module MyExtension
    module DatasetMethods
      def my_custom_method
           puts "foo"
      end
    end
  end
end

# Register the extension with Sequel
Sequel::Database.register_extension(:my_extension, Sequel::MyExtension)

```

But when I got error:

```
<internal:/usr/local/lib/ruby/3.3.0/rubygems/core_ext/kernel_require.rb>:37:in `require': cannot load such file -- sequel/extensions/my_extension (LoadError)
```

This happens immediately when I add a line like this in my sequel.rb:

```
Sequel::Database.register_extension(:my_extension, Sequel::MyExtension)
```

If you comment out this line and open rack-console there will be Sequel::MyExtension there

I don't understand which way to look? I don't see anything specific in my settings

**Comment by jeremyevans** [comment]
Thanks for posting about this. For a quick fix, if you add that `lib` directory to your load path, things will work.  The issue you are running into is that Database#extension calls Sequel.extension before doing anything, and that attempts to require the file.  Maybe this could be changed to check if the extension is already registered before calling Sequel.extension?  That would be more similar to how plugins operate, so it seems like a useful change.  I can work on that today.

**Reply by fbayaffo2097** [reply]
Thank! $LOAD_PATH.unshift File.expand_path('../../../lib', __FILE__) it's enough to check the work for now, I'll wait for your changes, thanks! 😸

**Reply by jeremyevans** [reply]
Fixed in c6c95889e9dab80e99e7f417ea85d097c743cecc
